src/Orc.Controls/Controls/StepBar/ViewModels/StepBarViewModel.cs

```diff
@@ -0,0 +1,121 @@
+
+        #region Constructors
+        public StepBarViewModel()
+        {
+            ContentLoaded = new TaskCommand<Views.StepBar>(ContentLoadedExecuteAsync, ContentLoadedCanExecute);
+            QuickNavigateToItem = new TaskCommand<IStepBarItem>(QuickNavigateToItemExecuteAsync, QuickNavigateToItemCanExecute);
+        }
+        #endregion Constructors
+
+        #region Properties
+        public IList<IStepBarItem> Items { get; set; }
+
+        public IStepBarItem SelectedItem { get; set; }
+
+        public bool AllowQuickNavigation { get; set; }
+        #endregion Properties
+
+        #region Methods
+        public virtual async Task MoveForwardAsync()
+        {
+            if (_currentIndex < Items.Count - 1)
+            {
+                Items[_currentIndex].IsVisited = true;
+                _currentIndex++;
+                SelectedItem = Items[_currentIndex];
+                Items[_currentIndex].IsVisited = true;
+            }
+        }
+
+        public virtual async Task MoveBackAsync()
+        {
+            if (_currentIndex > 0)
+            {
+                _currentIndex--;
+                SelectedItem = Items[_currentIndex];
+            }
+        }
+
+        protected internal async Task SetSelectedItem(int newIndex)
+        {
+            if (_currentIndex >= 0 && _currentIndex < Items.Count)
+            {
+                _currentIndex = newIndex;
+                SelectedItem = Items[_currentIndex];
+            }
+        }
+
+        public bool IsLastItem(IStepBarItem item)
+        {
+            return item.Equals(Items.LastOrDefault());
+        }
+        #endregion Methods
+
+        #region Commands
+        public TaskCommand<IStepBarItem> QuickNavigateToItem { get; private set; }
+
+        public bool QuickNavigateToItemCanExecute(IStepBarItem parameter)
+        {
+            if (!AllowQuickNavigation)
+            {
+                return false;
+            }
+
+            if (!parameter.IsVisited)
+            {
+                return false;
+            }
+
+            if (SelectedItem == parameter)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public async Task QuickNavigateToItemExecuteAsync(IStepBarItem parameter)
+        {
+            var item = parameter;
+            if (item != null && item.IsVisited && Items is List<IStepBarItem>)
+            {
+                var index = Items.IndexOf(item);
+
+                await SetSelectedItem(index);
+            }
+        }
+
+        public TaskCommand<Views.StepBar> ContentLoaded { get; private set; }
+
+        public bool ContentLoadedCanExecute(Views.StepBar parameter)
+        {
+            return true;
+        }
+
+        public async Task ContentLoadedExecuteAsync(Views.StepBar parameter)
+        {
+            if (parameter.Items != null)
+            {
+                AllowQuickNavigation = parameter.AllowQuickNavigation;
```

Comment:
Should be called AllowSelection. Please remember that this is a completely separate control from the wizard, which we might eventually use inside the wizard, but it should not be aware of this.